third_party/move/move-prover/boogie-backend/src/lib.rs

```diff
@@ -184,6 +184,27 @@ pub fn add_prelude(
         sh_instances = vec![];
         bv_instances = vec![];
     }
+
+    let mut all_types = mono_info
```

Comment:
Is this going to be very big?  Is mono_info every type in the program?

Reply:
For aptos-framework, there are about 500 uninterpreted functions generated while the generated boogie code is about 300k line. So IMHO as long as the program itself is reasonable, the number of types in the program is not a big concern.